src/internal/components/file-input/__tests__/file-input.test.tsx

```diff
@@ -0,0 +1,158 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { fireEvent, render as testingLibraryRender, screen } from '@testing-library/react';
+
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+import '../../../../__a11y__/to-validate-a11y';
+import InternalFileInput, { FileInputProps } from '../../../../../lib/components/internal/components/file-input';
+import FileInputWrapper from '../../../../../lib/components/test-utils/dom/internal/file-input';
+
+jest.mock('@cloudscape-design/component-toolkit/internal', () => ({
+  ...jest.requireActual('@cloudscape-design/component-toolkit/internal'),
+  warnOnce: jest.fn(),
+}));
+
+jest.mock('../../../../../lib/components/internal/utils/date-time', () => ({
+  formatDateTime: () => '2020-06-01T00:00:00',
+}));
+
+const onChange = jest.fn();
+
+afterEach(() => {
+  (warnOnce as jest.Mock).mockReset();
+  onChange.mockReset();
+});
+
+const defaultProps: FileInputProps = {
+  value: [],
+  onChange,
+};
+
+const file1 = new File([new Blob(['Test content 1'], { type: 'text/plain' })], 'test-file-1.txt', {
+  type: 'text/plain',
+  lastModified: 1590962400000,
+});
+const file2 = new File([new Blob(['Test content 2'], { type: 'text/plain' })], 'test-file-2.txt', {
+  type: 'image/png',
+  lastModified: 1590962400000,
+});
+
+function render(props: Partial<FileInputProps>) {
+  const renderResult = testingLibraryRender(
+    <div>
+      <InternalFileInput {...{ ...defaultProps, ...props }}>Choose files</InternalFileInput>
+      <div id="test-label">Test label</div>
+    </div>
+  );
+  const element = renderResult.container.querySelector<HTMLElement>(`.${FileInputWrapper.rootSelector}`)!;
+  return new FileInputWrapper(element)!;
+}
+
+describe('FileInput input', () => {
+  test('`multiple` property is assigned', () => {
+    render({ multiple: false });
```

Comment:
Again, is this needed? Multiple is already rendered as false below in the assertion